### ISSUE TITLE
"eager" instamine mode does not work in browser

### ISSUE THREAD

**Issue by MicaiahReid** [ConsenSys-archive/ganache#3308]
When using Ganache in the browser, the following line fails:
```TypeScript
if (options.miner.instamine === "eager") {
  // in eager instamine mode we must delay the broadcast of new blocks
  await new Promise(resolve => {
    // we delay emitting blocks and blockLogs because we need to allow for:
    // ```
    //  await provider.request({"method": "eth_sendTransaction"...)
    //  await provider.once("message") // <- should work
    // ```
    // If we don't have this delay here the messages will be sent before
    // the call has a chance to listen to the event.
    setImmediate(async () => {
      // emit block logs first so filters can pick them up before
      // block listeners are notified
      await Promise.all([
        this.emit("blockLogs", blockLogs),
        this.emit("block", block)
      ]);
      resolve(void 0);
    });
  });
}
```
With the error: `setImmediate is not defined`.

It looks like `setImmediate` needs to be polyfilled for "eager" mode to work properly in the browser.

**Comment by davidmurdoch** [comment]
https://github.com/trufflesuite/ganache/blob/279ab6dcd2ef5ac0a84ae928ca393b584c125c77/src/packages/ganache/index.ts#L23 🤔

**Comment by MicaiahReid** [comment]
Ooops, I was testing against the wrong branch! This was fixed in #3130
